integration-tests/splunk/src/test/java/org/apache/camel/quarkus/component/splunk/it/SplunkTest.java

```diff
@@ -16,138 +16,153 @@
+                () -> {
+                    String result = RestAssured.given()
+                            .contentType(ContentType.TEXT)
+                            .post("/splunk/results/savedSearch")
+                            .then()
+                            .statusCode(200)
+                            .extract().asString();
+
+                    return result.contains("Name: Sheldon" + suffix)
+                            && result.contains("Name: Leonard" + suffix)
+                            && result.contains("Name: Irma" + suffix);
+                });
     }
 
-    private void write(Map<String, String> data, String endpoint, String index) {
+    @Test
+    public void testStreamForRealtime() throws InterruptedException, ExecutionException {
+        String suffix = "_RealtimeSearchOfStream";
+        //there is a buffer for stream writing, therefore about 1MB of data has to be written into Splunk
+
+        //data are written in separated thread
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        //execute component server to wait for the result
+        Future futureResult = executor.submit(
+                () -> {
+                    for (int i = 0; i < 5000; i++) {
+                        write(suffix + i, ProducerType.STREAM, 100, false);
+                    }
+                });
 
-        String expectedResult = expectedResult(data);
+        try {
+            Awaitility.await().pollInterval(1000, TimeUnit.MILLISECONDS).atMost(60, TimeUnit.SECONDS).until(
+                    () -> {
+
+                        String result = RestAssured.given()
+                                .contentType(ContentType.TEXT)
+                                .post("/splunk/results/realtimeSearch")
+                                .then()
+                                .statusCode(200)
+                                .extract().asString();
+
+                        return result.contains("Name: Sheldon" + suffix)
+                                && result.contains("Name: Leonard" + suffix)
+                                && result.contains("Name: Irma" + suffix);
+                    });
+        } finally {
+            futureResult.cancel(true);
+        }
+    }
 
-        RestAssured.given()
+    private void write(String suffix, ProducerType producerType, int lengthOfRandomString, boolean raw) {
+        Consumer<Map> write = data -> RestAssured.given()
                 .contentType(ContentType.JSON)
-                .queryParam("index", index)
+                .queryParam("index", SplunkTestResource.TEST_INDEX)
                 .body(data)
-                .post("/splunk/" + endpoint)
+                .post("/splunk/write/" + producerType.name())
                 .then()
-                .statusCode(201)
-                .body(containsString(expectedResult));
+                .statusCode(201);
+
+        Map data[] = new Map[3];
```

Comment:
Nitpick - I assume this produces compiler warnings about usage of raw types? You can probably build the same data structure with `List.of` / `Map.of`.

Reply:
TBH I just thought, that it is not necessary to verify it, as the status code 201 means, that insert was done (the data is then verified in the tests, but I can add it back)

Reply:
Seems GitHub is not highlighting the bit of code I was referring to....

I mean the usage of `Map data[] = new Map[3];`.

Reply:
fixed